File: TransCoder_tokenized_test_set_functions/python/SMALLEST_ELEMENT_REPEATED_EXACTLY_K_TIMES_NOT_LIMITED_SMALL_RANGE.py
def smallestKFreq ( arr, n, k ) :
    mp = defaultdict ( lambda : 0 )
    for i in range ( n ) :
        mp [ arr [ i ] ] += 1
    res = sys . maxsize
    res1 = sys . maxsize
    for key, values in mp . items ( ) :
        if values == k :
            res = min ( res, key )
    return res if res != res1 else - 1

File: TransCoder_tokenized_test_set_functions/cpp/SMALLEST_ELEMENT_REPEATED_EXACTLY_K_TIMES_NOT_LIMITED_SMALL_RANGE.cpp
int smallestKFreq ( int a [ ], int n, int k ) {
  unordered_map < int, int > m;
  for ( int i = 0;
  i < n;
  i ++ ) m [ a [ i ] ] ++;
  int res = INT_MAX;
  for ( auto it = m . begin ( );
  it != m . end ( );
  ++ it ) if ( it -> second == k ) res = min ( res, it -> first );
  return ( res != INT_MAX ) ? res : - 1;
}